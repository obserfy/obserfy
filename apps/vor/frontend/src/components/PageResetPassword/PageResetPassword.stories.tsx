import { FC } from "react"
import PageResetPassword from "./PageResetPassword"

export default {
  title: "Page/PageResetPassword",
  component: PageResetPassword,
  parameters: {
    componentSubtitle: "Just a simple PageResetPassword",
  },
}

export const Basic: FC = () => <PageResetPassword token="sadfsadf" />
