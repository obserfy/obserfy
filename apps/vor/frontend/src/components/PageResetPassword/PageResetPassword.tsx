import { FC, FormEvent, useState } from "react"
import { Flex, Box, Button } from "theme-ui"
import { track } from "../../analytics"
import { Link } from "../Link/Link"

import { Typography } from "../Typography/Typography"
import Input from "../Input/Input"
import Icon from "../Icon/Icon"
import { ReactComponent as AlertIcon } from "../../icons/alert.svg"
import { ReactComponent as CheckmarkIcon } from "../../icons/checkmark.svg"

import { ReactComponent as NextIcon } from "../../icons/next-arrow.svg"
import LoadingIndicator from "../LoadingIndicator/LoadingIndicator"
import { doPasswordResetApi } from "../../hooks/api/doPasswordResetApi"
import Spacer from "../Spacer/Spacer"

interface Props {
  token: string
}
export const PageResetPassword: FC<Props> = ({ token }) => {
  const [password, setPassword] = useState("")
  const [retypePassword, setRetypePassword] = useState("")
  const [success, setSuccess] = useState(false)
  const [loading, setLoading] = useState(false)
  const [error, setError] = useState("")

  const isFormDisabled = success || loading

  async function handleSubmit(e: FormEvent): Promise<void> {
    e.preventDefault()
    // reset state
    setLoading(true)
    setError("")

    // Validate fields
    if (!password) {
      setError("New password can't be empty")
      setLoading(false)
      return
    }
    if (!retypePassword) {
      setError("Retyped password can't be empty")
      setLoading(false)
      return
    }
    if (retypePassword !== password) {
      setError("Your retyped password doesn't match.")
      setLoading(false)
      return
    }

    // Do the password reset
    const response = await doPasswordResetApi(token, password)
    if (response.status === 200) {
      setSuccess(true)
      track("Password reset success")
    } else {
      const body = await response.json()
      setError(body?.error?.message ?? "Something went wrong, please try again")
      track("Password reset failed")
    }

    setLoading(false)
  }

  return (
    <Flex
      sx={{
        justifyContent: "center",
        minHeight: "100vh",
        minWidth: "100vw",
      }}
      pt={3}
    >
      <Box
        as="form"
        p={3}
        sx={{ maxWidth: "maxWidth.sm", width: "100%" }}
        onSubmit={handleSubmit}
      >
        <Typography.H2 my={3}>Reset Password</Typography.H2>
        {!success && (
          <Typography.Body
            sx={{
              fontSize: 2,
            }}
            my={3}
            color="textMediumEmphasis"
          >
            Type in your new password and we&apos;ll get you up and running in
            no time :)
          </Typography.Body>
        )}

        {success && (
          <>
            <Flex
              my={3}
              py={2}
              px={3}
              backgroundColor="primaryLighter"
              sx={{
                alignItems: "center",
                borderWidth: 1,
                borderColor: "warning",
                borderRadius: "default",
              }}
            >
              <Icon
                sx={{ alignSelf: "flex-start" }}
                as={CheckmarkIcon}
                m={1}
                ml={0}
                fill="textPrimary"
              />
              <Typography.Body
                color="textPrimary"
                sx={{
                  width: "100%",
                  fontSize: 1,
                }}
                ml={1}
              >
                Password reset successful. You can try logging in with your new
                password.
              </Typography.Body>
            </Flex>
            <Flex>
              <Spacer />
              <Link to="/login">
                <Button variant="outline">
                  Login
                  <Icon as={NextIcon} mr={1} fill="textPrimary" />
                </Button>
              </Link>
            </Flex>
          </>
        )}

        {!success && (
          <>
            <Input
              sx={{ width: "100%" }}
              label="New Password"
              value={password}
              onChange={(e) => setPassword(e.target.value)}
              required
              mb={2}
              type="password"
              disabled={isFormDisabled}
            />
            <Input
              sx={{ width: "100%" }}
              label="Retype New Password"
              value={retypePassword}
              onChange={(e) => setRetypePassword(e.target.value)}
              required
              mb={3}
              type="password"
              disabled={isFormDisabled}
            />
            {error && (
              <Flex
                backgroundColor="tintError"
                sx={{ alignItems: "center", borderRadius: "default" }}
                my={3}
                py={2}
                px={3}
              >
                <Icon
                  sx={{ alignSelf: "flex-start" }}
                  as={AlertIcon}
                  m={2}
                  ml={0}
                  mr={2}
                  fill="onTintError"
                />
                <Typography.Body
                  sx={{
                    fontSize: 1,
                    width: "100%",
                  }}
                  color="onTintError"
                >
                  {error}
                </Typography.Body>
              </Flex>
            )}
            <Button
              variant="primaryBig"
              sx={{ width: "100%" }}
              disabled={isFormDisabled}
            >
              {loading && <LoadingIndicator color="onPrimary" />}
              Reset
            </Button>
          </>
        )}
      </Box>
    </Flex>
  )
}

export default PageResetPassword
