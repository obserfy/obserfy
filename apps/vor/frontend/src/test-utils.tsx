/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
