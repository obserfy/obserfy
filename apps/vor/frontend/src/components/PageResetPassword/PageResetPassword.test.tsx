import { render } from "../../test-utils"
import { Basic } from "./PageResetPassword.stories"

describe("PageResetPassword", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
