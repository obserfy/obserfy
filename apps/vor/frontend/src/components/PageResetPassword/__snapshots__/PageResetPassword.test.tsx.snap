// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageResetPassword should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-top: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  min-height: 100vh;
  min-width: 100vw;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 16px;
  max-width: 640px;
  width: 100%;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: bold;
  line-height: heading;
  color: var(--theme-ui-colors-text);
  text-rendering: optimizeLegibility;
  font-size: 64px;
  margin-top: 16px;
  margin-bottom: 16px;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin-top: 16px;
  margin-bottom: 16px;
  color: var(--theme-ui-colors-textMediumEmphasis);
  font-size: 16px;
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: var(--theme-ui-colors-textMediumEmphasis);
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: var(--theme-ui-colors-textMediumEmphasis);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-bottom: 4px;
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-8 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  width: auto;
  border-color: var(--theme-ui-colors-border);
  background-color: var(--theme-ui-colors-surface);
  font-size: 16px;
  padding: 12px;
  color: var(--theme-ui-colors-text);
  margin-bottom: 8px;
  --theme-ui-input-autofill-bg: var(--theme-ui-colors-background);
  width: 100%;
  width: 100%;
}

.emotion-8:autofill,
.emotion-8:autofill:hover,
.emotion-8:autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-8:autofill:first-line,
.emotion-8:autofill:hover:first-line,
.emotion-8:autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-8:-webkit-autofill,
.emotion-8:-webkit-autofill:hover,
.emotion-8:-webkit-autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-8:-webkit-autofill:first-line,
.emotion-8:-webkit-autofill:hover:first-line,
.emotion-8:-webkit-autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-8:hover,
.emotion-8:focus {
  border-color: var(--theme-ui-colors-primary);
}

.emotion-8::-webkit-input-placeholder {
  font-style: italic;
}

.emotion-8::-moz-placeholder {
  font-style: italic;
}

.emotion-8:-ms-input-placeholder {
  font-style: italic;
}

.emotion-8::placeholder {
  font-style: italic;
}

.emotion-8:disabled {
  opacity: 0.68;
  border-color: var(--theme-ui-colors-border);
  color: var(--theme-ui-colors-textDisabled);
}

.emotion-12 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  width: auto;
  border-color: var(--theme-ui-colors-border);
  background-color: var(--theme-ui-colors-surface);
  font-size: 16px;
  padding: 12px;
  color: var(--theme-ui-colors-text);
  margin-bottom: 16px;
  --theme-ui-input-autofill-bg: var(--theme-ui-colors-background);
  width: 100%;
  width: 100%;
}

.emotion-12:autofill,
.emotion-12:autofill:hover,
.emotion-12:autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-12:autofill:first-line,
.emotion-12:autofill:hover:first-line,
.emotion-12:autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-12:-webkit-autofill,
.emotion-12:-webkit-autofill:hover,
.emotion-12:-webkit-autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-12:-webkit-autofill:first-line,
.emotion-12:-webkit-autofill:hover:first-line,
.emotion-12:-webkit-autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-12:hover,
.emotion-12:focus {
  border-color: var(--theme-ui-colors-primary);
}

.emotion-12::-webkit-input-placeholder {
  font-style: italic;
}

.emotion-12::-moz-placeholder {
  font-style: italic;
}

.emotion-12:-ms-input-placeholder {
  font-style: italic;
}

.emotion-12::placeholder {
  font-style: italic;
}

.emotion-12:disabled {
  opacity: 0.68;
  border-color: var(--theme-ui-colors-border);
  color: var(--theme-ui-colors-textDisabled);
}

.emotion-13 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-onPrimary);
  background-color: var(--theme-ui-colors-primary);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: rgba(27, 31, 35, 0.1) 0px 1px 0px 0px,rgba(255, 255, 255, 0.03) 0px 1px 0px 0px inset;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  width: 100%;
}

.emotion-13:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-13:hover,
.emotion-13:focus {
  background-color: var(--theme-ui-colors-primaryDark);
}

.emotion-13:hover:disabled,
.emotion-13:focus:disabled {
  background-color: var(--theme-ui-colors-primary);
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <form
        class="emotion-2"
      >
        <h2
          class="emotion-3"
        >
          Reset Password
        </h2>
        <p
          class="emotion-4"
        >
          Type in your new password and we'll get you up and running in no time :)
        </p>
        <label
          class="emotion-5"
        >
          <div
            class="emotion-6"
          >
            New Password
          </div>
          <div
            class="emotion-7"
          >
            <input
              class="emotion-8"
              required=""
              type="password"
              value=""
            />
          </div>
        </label>
        <label
          class="emotion-5"
        >
          <div
            class="emotion-6"
          >
            Retype New Password
          </div>
          <div
            class="emotion-7"
          >
            <input
              class="emotion-12"
              required=""
              type="password"
              value=""
            />
          </div>
        </label>
        <button
          class="emotion-13"
        >
          Reset
        </button>
      </form>
    </div>
  </div>
</div>
`;
